src/androidTest/java/de/blau/android/layer/LayerDialogTest.java

```diff
@@ -262,4 +264,35 @@ public void backgroundLayer() {
         assertNotNull(layer);
         assertEquals(TileLayerSource.LAYER_MAPNIK, layer.getTileLayerConfiguration().getId());
     }
+
+    /**
+     * Checks if filters are correct
+     */
+    @Test
+    public void layerFilter() {
+        assertTrue(TestUtils.clickResource(device, true, device.getCurrentPackageName() + ":id/layers", true));
+        assertTrue(TestUtils.clickButton(device, device.getCurrentPackageName() + ":id/add", true));
+        assertTrue(TestUtils.clickText(device, false, main.getString(R.string.menu_layers_add_backgroundlayer), true, false));
```

Comment:
After this you should add an, not asserted, click on the OK button to click away the Tip dialog if it is displayed (which will happen if you are just running this test), see the backgroundLayer test.